### ISSUE TITLE
Github action does not seem to work

### ISSUE THREAD

**Issue by kcross-ctoken** [lukaszraczylo/semver-generator#50]
**Describe the bug**
The action fails with no output

**Output with debug**
none

**Expected behavior**
to at least put error output

**Additional context**
I would suggest using the github action in your own release scripts to check that it works for creating the release versions. I.e. dogfood your release in the release scripts instead of 
```
         DOWNLOAD_URL=$(curl -s https://api.github.com/repos/lukaszraczylo/semver-generator/releases/latest | grep -E ".*browser_download_url.*linux-" | grep -vE "(arm64|md5)" \
            | cut -d '"' -f 4)
          curl -s -L -o semver-gen "$DOWNLOAD_URL" && chmod +x semver-gen
          TMP_RELEASE_VERSION=$(./semver-gen generate -l -c config-release.yaml | sed -e 's|SEMVER ||g')
```
Since this is the only why it seems to work Im now having to use this instead of the github action.

**Comment by lukaszraczylo** [comment]
Thank you for reporting, having a look now.

**Comment by lukaszraczylo** [comment]
@kcross-ctoken how do you use the github action? I've re-ran it on one of the repositories:

https://github.com/lukaszraczylo/go-simple-graphql/actions/runs/10263942990/job/28396864774

```
      - name: Establish semver
        id: semver
        uses: lukaszraczylo/semver-generator@v1
        with:
          config_file: semver.yaml
          repository_local: true
          github_username: ${{ github.ACTOR }}
          github_token: $GHCR_TOKEN

      - name: Semver check
        run: |
          echo "Semantic version detected: ${{ steps.semver.outputs.semantic_version }}"
```

You need to create semver.yaml file in the repository root, and it just works :)

**Comment by kcross-ctoken** [comment]
hmmm let me try again

**Comment by kcross-ctoken** [comment]
when I put the repository as    
`repository_url: ${{ github.server_url }}/${{ github.repository }}` it fails with no errors or outputs 
if I use repository_local: true it comes back with 0.0.0
If I use the command on the same repo locally `./semver-gen generate -l -c semver.yaml -e | sed -e 's|SEMVER ||g'  it comes back with `0.0.23`


Im using this atm:

```yaml
      - name: semver config
        uses: "DamianReeves/write-file-action@master"
        with:
          path: semver.yaml
          write-mode: overwrite
          contents: |
            version: 1
            wording:
              minor:
                - feat
              major:
                - breaking

      - name: Establish semver
        id: semver
        uses: lukaszraczylo/semver-generator@v1
        with:
          config_file: semver.yaml
          repository_local: true
          #repository_url: ${{ github.server_url }}/${{ github.repository }}
          github_username: ${{ github.ACTOR }}
          github_token: ${{secrets.REPO_RO_TOKEN}}
          strict: false
          existing: true
```

**Comment by kcross-ctoken** [comment]
It would be nice to be able to write the config.yaml directly into the yaml using =| ... but thats a feature request.

**Comment by lukaszraczylo** [comment]
I understand the issue now.
If you create repository, and want to use semver generator on it - just create the semver.yaml file.

The `repository_url` feature is an ancient and testing-only one used in tests only allowing for the tests to run against the mock repositories I've created to verify that the version is calculated correctly. I will need to update the README in this case and most likely sunset the repository_url as I admit now - it's a bit confusing :)

**Comment by kcross-ctoken** [comment]
ok. Im using it in a github actions library so I have a use clause which means I can have the config yaml locally as its not guaranteed that the using repo will have a config file locally.

**Comment by lukaszraczylo** [comment]
Have you tried the `semver-gen generate -r https://github.com/org/repository -c semver.yaml -e --debug` ?

**Comment by kcross-ctoken** [comment]
In github run action
```
++ ./semver-gen generate -l -r https://github.com/SomeWhere/some-repo -c semver.yaml -e --debug
++ sed -e 's|SEMVER ||g'
+ TMP_RELEASE_VERSION=0.0.1
+ echo '>> Release version: 0.0.1 <<'
+ echo RELEASE_VERSION=0.0.1
>> Release version: 0.0.1 <<
```

where locally on the same repo 
```
./semver-gen generate -l -e -c semver.yaml -e --debug
SEMVER 0.0.23
```
Which is what I was expecting 🤔

**Comment by lukaszraczylo** [comment]
Remove the `-l` from the first one - it points the generator towards the `local` repository

**Comment by kcross-ctoken** [comment]
ok finally Ive got a reasonable result.
```
++ ./semver-gen generate -r https://github.com/ClearToken/util-data -c semver.yaml -e --debug
+ TMP_RELEASE_VERSION=0.0.18
+ echo '>> Release version: 0.0.18 <<'
+ echo RELEASE_VERSION=0.0.18
>> Release version: 0.0.18 <<
```
Im guessing the 18 instead of 23  is because its not looking at the current commits on the branch Im working on more whats on main ?

**Comment by lukaszraczylo** [comment]
Yep, it always defaults back to the main / master :)

**Comment by lukaszraczylo** [comment]
@kcross-ctoken just as a follow-up - all good? :)

**Comment by kcross-ctoken** [comment]
hiya ... yeah got it going but It would have saved me a lot of time with these:
1. the repository_url does not output any errors and fails.
2. adding  `fetch-depth: 0` on the checkout is imperative to get the right semver version (not 0.0.0) too.
3. I have to write the file locally instead of being able to pipe it in via string 🤷 ... could be nice to have the config visible since its not large.

Thanks for your support
